file-ballerina/src/file/filepath/file_path.bal

```diff
@@ -0,0 +1,462 @@
+# ```ballerina
+#  boolean|file:Error isAbsolute = file:isAbsolutePath("/A/B/C");
+# ```
+#
+# + path - String value of the file path
+# + return - `true` if path is absolute, `false` otherwise, or else an `file:Error`
+#            occurred if the path is invalid
+public isolated function isAbsolutePath(string path) returns boolean|Error {
+    if (path.length() <= 0) {
+        return false;
+    }
+    if (isWindows) {
+        return check getVolumnNameLength(path) > 0;
+    } else {
+        return check charAt(path, 0) == "/";
+    }
+}
+
+# Retrieves the base name of the file from the provided location,
+# which is the last element of the path.
+# Trailing path separators are removed before extracting the last element.
+# ```ballerina
+#  string|file:Error name = file:basename("/A/B/C.txt");
+# ```
+#
+# + path - String value of file path
+# + return - The name of the file or else a `file:Error` if the path is invalid
+public isolated function basename(string path) returns string|Error {
+    string validatedPath = check parse(path);
+    int[] offsetIndexes = check getOffsetIndexes(validatedPath);
+    int count = offsetIndexes.length();
+    if (count == 0) {
+        return "";
+    }
+    if (count == 1 && validatedPath.length() > 0) {
+        if !(check isAbsolutePath(validatedPath)) {
+            return validatedPath;
+        }
+    }
+    int lastOffset = offsetIndexes[count - 1];
+    return validatedPath.substring(lastOffset, validatedPath.length());
+}
+
+# Returns the enclosing parent directory.
+# If the path is empty, parent returns ".".
+# The returned path does not end in a separator unless it is the root directory.
+# ```ballerina
+#  string|file:Error parentPath = file:parentPath("/A/B/C.txt");
+# ```
+#
+# + path - String value of the file/directory path
+# + return - Path of the parent directory or else a `file:Error`
+#            if an error occurred while getting the parent directory
+public isolated function parentPath(string path) returns string|Error {
+    string validatedPath = check parse(path);
+    int[] offsetIndexes = check getOffsetIndexes(validatedPath);
+    int count = offsetIndexes.length();
+    if (count == 0) {
+        return "";
+    }
+    int len = offsetIndexes[count-1] - 1;
+    if (len < 0) {
+        return "";
+    }
+    int offset;
+    string root;
+    [root, offset] = check getRoot(validatedPath);
+    if (len < offset) {
+        return root;
+    }
+    return validatedPath.substring(0, len);
+}
+
+# Returns the shortest path name equivalent to the given path.
+# Replace the multiple separator elements with a single one.
+# Eliminate each "." path name element (the current directory).
+# Eliminate each inner ".." path name element (the parent directory).
+# ```ballerina
+#  string|file:Error normalizedPath = file:normalizePath("foo/../bar", file:CLEAN);
+# ```
+#
+# + path - String value of the file path
+# + option - Normalization option
```

Comment:
It's better if we could be a little more descriptive about the options. If the `NormOption` enum is documented, then fine.